src/celeritas/optical/model/RayleighModel.cc

```diff
@@ -59,23 +59,14 @@ RayleighModel::RayleighModel(ActionId id, SPConstImported imported, Input input)
 
     for (auto mat : range(OptMatId(imported_.num_materials())))
     {
-        if (input_)
+        if (!(imported_.mfp(mat)
+              || (input_ && input_.imported_materials->rayleigh(mat))))
         {
-            CELER_VALIDATE(
-                imported_.mfp(mat) || input_.imported_materials->rayleigh(mat),
-                << "Rayleigh model requires either imported MFP or "
-                   "material parameters to build MFPs for each optical "
-                   "material");
-        }
-        else
-        {
-            CELER_VALIDATE(imported_.mfp(mat),
-                           << "Rayleigh model requires imported MFP for each "
-                              "optical material");
+            CELER_LOG(debug)
+                << "Rayleigh model has no MFP data for material " << mat.get();
```

Comment:
This log message is redundant with the one in the constructor; I'd delete this one to remove the complex conditionals.